core/src/main/java/apoc/schema/Schemas.java

```diff
@@ -142,65 +149,73 @@ private AssertSchemaResult createUniqueConstraint(Schema schema, String lbl, Str
     }
 
     public List<AssertSchemaResult> assertIndexes(Map<String, List<Object>> indexes0, boolean dropExisting) throws ExecutionException, InterruptedException, IllegalArgumentException {
-        Schema schema = tx.schema();
-        Map<String, List<Object>> indexes = copyMapOfObjects(indexes0);
-        List<AssertSchemaResult> result = new ArrayList<>(indexes.size());
-
-        for (IndexDefinition definition : schema.getIndexes()) {
-            if (!definition.isNodeIndex())
-                continue;
-            if (definition.getIndexType() == IndexType.LOOKUP)
-                continue;
-            if (definition.isConstraintIndex())
-                continue;
-
-            String label = Iterables.single(definition.getLabels()).name();
-            List<String> keys = new ArrayList<>();
-            definition.getPropertyKeys().forEach(keys::add);
-
-            AssertSchemaResult info = new AssertSchemaResult(label, keys);
-            if(indexes.containsKey(label)) {
-                if (keys.size() > 1) {
-                    indexes.get(label).remove(keys);
-                } else if (keys.size() == 1) {
-                    indexes.get(label).remove(keys.get(0));
-                } else
-                    throw new IllegalArgumentException("Label given with no keys.");
-            }
+        // to prevent transaction locks (e.g. with lookup index and assert with creation schema)
```

Comment:
in che senso?

Reply:
Perché ho notato che se non apro un'altra transazione, se creo un `LOOKUP INDEX` la query rimane in pending.
Non credo sia una questione di implementazione sbagliata visto che lo fa solo con i Lookup, gli altri funzionano anche come stava prima senza `beginTx()`